src/Carousel.vue

```diff
@@ -622,28 +731,46 @@ export default {
       );
       return slides[index];
     },
+    /**
+     * Calculates the offset for the page
+     * @return {Number} The amount of offset
+     */
+    getOffset(page) {
+      return this.scrollPerPage && !this.infinite
+        ? Math.min(
+            this.slideWidth * this.currentPerPage * page,
+            !this.infinite ? this.maxOffset : this.maxOffset + this.slideWidth
+          )
+        : Math.min(
+            this.slideWidth * page,
+            !this.infinite ? this.maxOffset : this.maxOffset + this.slideWidth
+          );
+    },
     /**
      * Set the current page to a specific value
      * This function will only apply the change if the value is within the carousel bounds
      * @param  {Number} page The value of the new page number
      */
     goToPage(page) {
-      if (page >= 0 && page <= this.pageCount) {
-        this.offset = this.scrollPerPage
-          ? Math.min(
-              this.slideWidth * this.currentPerPage * page,
-              this.maxOffset
-            )
-          : Math.min(this.slideWidth * page, this.maxOffset);
-
-        // restart autoplay if specified
-        if (this.autoplay && !this.autoplayHoverPause) {
-          this.restartAutoplay();
-        }
+      // Small timeout is needed for the infinity effect
+      setTimeout(
```

Comment:
The problem with that was that if I did that, the goToPage would recieve a null/undefined directly.
Since I didn't have the time to make pretty much everything async to solve that, I solved it with a timeout.
Again, my own personal time was an issue :)

I'd like to hear more opinions though, or maybe someone that feels brave enough to change it all to async